apis/rng/src/lib.rs

```diff
@@ -0,0 +1,89 @@
+#![no_std]
+
+use core::cell::Cell;
+
+use libtock_platform::{share, AllowRw, DefaultConfig, ErrorCode, Subscribe, Syscalls};
+pub struct Rng<S: Syscalls>(S);
+
+impl<S: Syscalls> Rng<S> {
+    /// Returns Ok() if the driver was present.This does not necessarily mean
+    /// that the driver is working.
+    pub fn exists() -> Result<(), ErrorCode> {
+        S::command(DRIVER_NUM, EXISTS, 0, 0).to_result()
+    }
+
+    /// Register a listener to be called when the random generation is finished
+    pub fn register_listener<'share>(
+        listener: &'share Cell<Option<(u32, u32)>>,
```

Comment:
What do that two `u32` values from the listener mean?

Reply:
The first one is always 0, and the second one represents the number of random bytes successfuly written in the buffer.